theano/gpuarray/basic_ops.py

```diff
@@ -1245,8 +1245,27 @@ class GpuJoin(HideC, Join):
 
     """
     _f16_ok = True
+    __props__ = ("view",)
     params_type = gpu_context_type
 
+    def __init__(self, view=-1):
+        self.view = view
+        if view != -1:
+            # since the first input is always the axis, the tensors
+            # start from index 1.
+            self.view_map = {0: [1 + view]}
+
+    # def __str__(self):
+    #     if self.view == -1:
+    #         return "Join"
+    #     else:
+    #         return super(Join, self).__str__()
+
+    # def __setstate__(self, d):
+    #     self.__dict__.update(d)
+    #     if not hasattr(self, "view"):
+    #         self.view = -1
```

Comment:
Remove those 2 commanted method. They are inherited. Go modify the parent \__str__ method to use self.__class__.__name__ instead of "Join". This will use the good string Join or GpuJoin depending of the class.